src/main/java/org/jabref/migrations/PreferencesMigrations.java

```diff
@@ -436,7 +436,7 @@ static void changeColumnVariableNamesFor51(JabRefPreferences preferences) {
      * they can deal with.
      */
     static void restoreVariablesForBackwardCompatibility(JabRefPreferences preferences) {
-        List<String> oldColumnNames = preferences.getStringList(JabRefPreferences.COLUMN_NAMES);
```

Comment:
Oh, why this? It is good to use string constants instead of hard-typed strings.

Reason: Ctrl+Click works on a constant - and not on a string.

We cannot add this functionality to IDEs easily, thus we need to adapt our coding behavior to IDEs cacpabilities.

Reply:
In this method, we are checking if the old preferences key from version 5.0, "columnNames" exists. 
JabRefPreferences.COLUMN_NAMES= "mainTableColumnNames", is the new name.

Reply:
Just add a comment above:

Check for 5.x prefernce name"...". The new one is {@link ...}